labelbox/schema/annotation_import.py

```diff
@@ -454,3 +454,154 @@ def _create_mea_import_from_bytes(
+        return cls._create_label_import_from_bytes(client, project_id, name,
+                                                   data, len(data))
+
+    @classmethod
+    def create_from_url(cls, client: "labelbox.Client", project_id: str,
+                        name: str, url: str) -> "LabelImport":
+        """
+        Create an label annotation import job from a url
+        The url must point to a file containing label annotations.
+
+        Args:
+            client: Labelbox Client for executing queries
+            project_id: Project to import labels into
+            name: Name of the import job. Can be used to reference the task later
+            url: Url pointing to file to upload
+        Returns:
+            LabelImport
+        """
+        if requests.head(url):
+            query_str = cls._get_url_mutation()
+            return cls(
+                client,
+                client.execute(query_str,
+                               params={
+                                   "fileUrl": url,
+                                   "projectId": project_id,
+                                   'name': name
+                               })["createLabelImport"])
+        else:
+            raise ValueError(f"Url {url} is not reachable")
+
+    @classmethod
+    def from_name(cls,
+                  client: "labelbox.Client",
+                  project_id: str,
+                  name: str,
+                  as_json: bool = False) -> "LabelImport":
+        """
+        Retrieves an label import job.
+
+        Args:
+            client: Labelbox Client for executing queries
+            project_id:  ID used for querying import jobs
+            name: Name of the import job.
+        Returns:
+            LabelImport
+        """
+        query_str = """query getLabelImportPyApi($projectId : ID!, $name: String!) {
+            labelImport(
+                where: {projectId: $projectId, name: $name}){
+                    %s
+                }}""" % query.results_query_part(cls)
+        params = {
+            "projectId": project_id,
+            "name": name,
+        }
+        response = client.execute(query_str, params)
+        if response is None:
+            raise labelbox.exceptions.ResourceNotFoundError(LabelImport, params)
+        response = response["labelImport"]
+        if as_json:
+            return response
+        return cls(client, response)
+
+    @classmethod
+    def _get_url_mutation(cls) -> str:
+        return """mutation createLabelImportPyApi($projectId : ID!, $name: String!, $fileUrl: String!) {
+            createLabelImport(data: {
+                projectId: $projectId
+                name: $name
+                fileUrl: $fileUrl
+            }) {%s}
+        }""" % query.results_query_part(cls)
+
+    @classmethod
+    def _get_file_mutation(cls) -> str:
+        return """mutation createLabelImportPyApi($projectId : ID!, $name: String!, $file: Upload!, $contentLength: Int!) {
```

Comment:
Maybe we shoudn't name these the same. It might make debugging easier. (Same with the mal and mea ones too.)

Reply:
sure!